### DISCUSSION TITLE
UDP问题

### DISCUSSION THREAD

**Discussion by Murasame233** [aiqinxuancai/discord-proxy#20] [Ideas]
跟朋友屏幕共享，发现 discord在使用 udp 获取屏幕共享，朋友在国外开共享给我看，我卡。
好像你们没做udp代理

**Comment by aiqinxuancai** [comment]
目前的实现上理论上是没有UDP代理的

**Comment by acking-you** [comment]
> 目前的实现上理论上是没有UDP代理的

挺好奇这个dll是怎么被劫持的，毕竟discord原先的目录中应该不存在version.dll。
我的理解是，原理是通过这个dll劫持，把系统HTTP代理的环境变量的值直接在该进程中指定，可能这个环境变量的值是HTTP_PROXY 和 HTTPS_PROXY。

**Comment by aiqinxuancai** [comment]
> > 目前的实现上理论上是没有UDP代理的
> 
> 挺好奇这个dll是怎么被劫持的，毕竟discord原先的目录中应该不存在version.dll。 我的理解是，原理是通过这个dll劫持，把系统HTTP代理的环境变量的值直接在该进程中指定，可能这个环境变量的值是HTTP_PROXY 和 HTTPS_PROXY。

你的理解是没错的，version.dll是一个常用的系统Dll，只要Dll存在于exe的导入表中，根据windows的导入表加载机制，会优先当前目录寻找dll，如果找不到才会到系统目录中进行加载，所以会加载我们实现的dll。

**Comment by acking-you** [comment]
> > > 目前的实现上理论上是没有UDP代理的
> > 
> > 
> > 挺好奇这个dll是怎么被劫持的，毕竟discord原先的目录中应该不存在version.dll。 我的理解是，原理是通过这个dll劫持，把系统HTTP代理的环境变量的值直接在该进程中指定，可能这个环境变量的值是HTTP_PROXY 和 HTTPS_PROXY。
> 
> 你的理解是没错的，version.dll是一个常用的系统Dll，只要Dll存在于exe的导入表中，根据windows的导入表加载机制，会优先当前目录寻找dll，如果找不到才会到系统目录中进行加载，所以会加载我们实现的dll。

好方式，学到了！感谢👍
